documentation/writing_plans.md

```diff
@@ -173,6 +173,75 @@ parameterized `Sensitive` type, such as `Sensitive` or `Sensitive[Hash]`. Other
 types, such as `Optional[Sensitive]` or `Hash[String, Sensitive]`, will not be
 automatically masked.
 
+## Using Hiera data in plans
+
+You can also use the Puppet `lookup()` function in plans to lookup Hiera data. It's useful to think
+of looking up Hiera data in Bolt plans in two different contexts: inside [apply
+blocks](applying_manifest_blocks.md), and outside apply blocks. Inside apply blocks Bolt is
+compiling Puppet catalogs in a per-target context, and has unfettered access to Hiera data. You can
+use the same Hiera config and data you would for Puppet, and `lookup()` data as you would expect
+inside apply blocks. Outside apply blocks, Bolt is essentially executing a script. It doesn't have a
+concept of a target or context, and thus cannot load per-target data. This breaks common Hiera
+features like interpolations.
+
+In order to allow users to use their same Hiera configurations with Bolt to look up per-target data
+inside apply blocks in addition to being able to look up static data outside of apply blocks we
+created the `plan_hierarchy`. `plan_hierarchy` is a key you specify in your Hiera configuration file
+at the same level as `hierarchy` which loads static data that can be looked up by Bolt outside of
+apply blocks. If present, Bolt uses the data under the `plan_hierarchy` key outside of apply
+blocks, and uses the `hierarchy` key inside of apply blocks. This allows you to use your existing
+Hiera configs in Bolt plans without encountering an error if interpolation exists and your plan
+tries to look up data outside of an apply block.
```

Comment:
```suggestion
To look up static data outside of apply blocks, you can add a `plan_hierarchy` key to your Hiera
configuration. The `plan_hierarchy` key is specified at the same level as the `hierarchy` key
and is used whenever you look up data outside an apply block. By specifying both keys in the
same Hiera configuration, you can look up data inside and outside apply blocks in the same
plan. This allows you to use your existing Hiera configuration in Bolt plans without encountering
an error if interpolations exist and your plan tries to look up data outside an apply block.
```

Mostly trying to avoid "we created the `plan_hierarchy`" here.